proc/tool/analysis/analysis.cpp

```diff
@@ -0,0 +1,238 @@
+#include "include/analysis.h"
```

Comment:
I know you have args listed below, but it's probably best if it's up in a header